util/src/main/scala/com/scalableminds/util/io/ZipIO.scala

```diff
@@ -100,22 +100,27 @@ object ZipIO {
 
     def isFileHidden(e: ZipEntry): Boolean = new File(e.getName).isHidden || e.getName.startsWith("__MACOSX")
 
-    def stripPathFrom(path: Path, string: String): Option[String] = {
-      if(path.toString.contains(string))
-        Some(path.toString.substring(0, path.toString.indexOf(string)))
-      else
-        None
+    def stripPathFrom(path: Path, string: String): Option[Path] = {
+      for (i <- 0 until path.getNameCount) {
+
+        if (path.subpath(i, i + 1).toString.contains(string))
```

Comment:
Couldn't we use `== string` now?

Reply:
No because if there are multiple layers of the same category which are named e.g. `color_1`, `color_2`...
So in the case that one of those layers gets copied and is the only layer in the new dataset, this wouldn't get stripped.

Reply:
Maybe just add a comment that shows the possible problematic folder-structures. I think there's quite some invisible knowledge in those lines, that can only be figured out by looking at several datasets.